packages/edit-site/src/hooks/use-theme-style-variations/use-theme-style-variations-by-property.js

```diff
@@ -47,17 +47,31 @@ export function removePropertyFromObject( object, property ) {
 	return object;
 }
 
+/**
+ * Checks whether a style variation is empty.
+ *
+ * @param {Object} variation          A style variation object.
+ * @param {string} variation.title    The title of the variation.
+ * @param {Object} variation.settings The settings of the variation.
+ * @param {Object} variation.styles   The styles of the variation.
+ * @return {boolean} Whether the variation is empty.
+ */
+function isEmptyStyleVariation( { title, settings, styles } ) {
+	return (
+		title !== __( 'Default' ) && // Always preserve the default variation.
+		Object.keys( settings ).length === 0 &&
+		Object.keys( styles ).length === 0
+	);
+}
```

Comment:
Do you want to rename instead of changing the function then? I think it would be simpler as: `hasThemeVariation`

Then you can leave the previous `||` lines as before so we're not making a functional change here too.

```suggestion
function hasThemeVariation( { title, settings, styles } ) {
	return (
		title !== __( 'Default' ) || // Always preserve the default variation.
		Object.keys( settings ).length > 0 ||
		Object.keys( styles ).length > 0
	);
}
```